draft-ietf-tls-esni.md

```diff
@@ -1252,16 +1238,83 @@ TCP connections an attacker can open.
+not using the mechanism. To that end, ECH is designed to resemble a standard
+TLS handshake as much as possible. The most obvious difference is the extension
+itself: as long as middleboxes ignore it, as required by {{!RFC8446}}, the rest
+of the handshake is designed to look very much as usual.
+
+The GREASE ECH protocol described in {{grease-ech}} provides a low-risk way to
+evaluate the deployability of ECH. It is designed to mimic the real ECH protocol
+({{real-ech}}) without changing the security properties of the handshake. The
+underlying theory is that if GREASE ECH is deployable without triggering
+udifferentiable behavior, and real ECH is indifferentiable from GREASE ECH, then
+ECH should be deployable as well. Thus, our strategy for mitigating network
+ossification has two parts:
+1. Design a "cover" protocol (i.e., GREASE ECH) that meets some notion of
+   deployability.
+1. Design the "target" protocol (i.e., ECH) so that networks do not differentiate
+   between it and the cover protocol.
+
+This strategy is analogous to the methodology that lead to "Middlebox
+Compatibility Mode" for TLS 1.3 described in {{!RFC8446}}.
+
+In the first step, the deployability of GREASE ECH is established by deploying
+the protocol and measuring its impact. [[OPEN ISSUE: Run an experiment that
+compares the error rate of GREASE ECH connections with vanilla TLS 1.3.]]
+
+From this experiment we can deduce the set of features of the TLS handshake that
+are known to trigger differentiable treatment by the network. This set might
+include the plaintext extensions offered by the client or server, plaintext
+alert messages, the number of round trips over the network, the length of each
+message, and so on. We will call these the protocol's "observable features". Our
+goal is to design real ECH so that each execution of the protocol has the same
+observable features as if GREASE ECH were being executed instead. For example,
+if extensions are an observable feature, then regardless of whether they are
+running the target protocol or the cover protocol, the client and server should
+appear to negotiate the same set of extensions. That way the real protocol does
+not trigger differentiable treatment that would not have been triggered by the
+cover protocol.
+
+We say that the target protocol is "indifferentiable" from cover protocol if,
+when executed in the presence of the adversary, both protocols have the same set
+of observable features. The strength of this model depends on how rich the set
+of observable features is: it may include the transcript of the protocol's
+execution itself, yielding a very strong attacker; or it may be as coarse
+grained as the number of IP packets sent. The model's strength also depends on
+whether the attacker is passive or active (see {{goals}}); middleboxes have been
+known to exhibit active behavior, even without malicious intent [[TOOD: Add
+references]].
```

Comment:
```suggestion
whether the attacker is passive or active (see {{goals}}).
```

I'd suggest dropping this unless we can point to some evidence. Maybe @davidben has one?

Reply:
There are some amazing examples [here](https://mailarchive.ietf.org/arch/msg/tls/i9blmvG2BEPf1s1OJkenHknRw9c/):
> We found that Firepower middleboxes in "Decrypt - Resign" mode
terminate TLS connections, but do not send a compliant ClientHello:
They modify the original ClientHello to remove unknown ciphersuites,
EMS, and NPN, but incorrectly forward most other fields from the
original ClientHello, including unknown extensions (supported_versions
and key_shares), and the client random. This breaks TLS 1.3 servers.
Additionally, these devices forward the server random rather than
generating their own (which will break when deploying the TLS 1.3
anti-downgrade feature), and forward unknown signature algorithms
(which will break when deploying, e.g., Ed25519).

Reply:
Can we just cite this email? That would be fine for me!

Reply:
Cited references in Section D.4 of RFC 8446.